akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -414,13 +414,18 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
       joinSeedNodes(newSeedNodes)
     case msg: SubscriptionMessage ⇒
       publisher forward msg
+    case Welcome(from, gossip) ⇒
+      log.info("Accepting Welcome message (while uninitialized) from previous joinSeedNodeProcess initiated Join, from: [{}]", from)
```

Comment:
logging is already done in `welcome`

Reply:
It might be good to make it more obvious in the logging that it is somewhat unexpected to receive a Welcome message in this state - OK though.

Reply:
Yeah was my intent to log more here, but I guess the reply usually will come simply from a retry so it should be fine. (That’s what we saw and why we are touching this code now)